kcidb/test_orm.py

```diff
@@ -498,3 +500,637 @@ def test_repr(self):
+        )['test']
+        self.assertIn(
+            template_dict('test', {
+                "waived": False,
+                "start_time": "2021-11-23T03:52:13.666000+00:00",
+                "path": "baseline.dmesg.crit",
+                "status": "PASS",
+                "id": "kernelci:kernelci.org:619c656de1fb4af479f2efac",
+                "build_id": "kernelci:kernelci.org:619c64b171"
+                            "2847eccbf2efac",
+                "origin": "kernelci"
+            }),
+            test_of_origin_kernelci_and_redhat
+        )
+        self.assertIn(
+            template_dict('test', {
+                "waived": False,
+                "start_time": "2021-11-23T03:52:13.666000+00:00",
+                "path": "baseline.dmesg.crit",
+                "status": "PASS",
+                "id": "kernelci:kernelci.org:619c656de1fb4af479f2efac",
+                "build_id": "kernelci:kernelci.org:619c64b171"
+                            "2847eccbf2efac",
+                "origin": "kernelci"
+            }),
+            test_of_origin_kernelci_and_redhat
+        )
+
+        self.assertEqual(
+            self.source.oo_query(
+                kcidb.orm.Pattern.parse(
+                    '>test[kernelci:'
+                    'kernelci.org:619c656de1fb4af479f2efaa]<build#'
+                )
+            ),
+            {
+                "build": [
+                    template_dict('build', {
+                        "id": "kernelci:kernelci.org:619c64b1712847eccbf2efac",
+                        "origin": "kernelci",
+                        "checkout_id": "_:kernelci:82bcf49e5e6ad57"
+                                       "0ff61ffcd210cf85c5ec8d896"
+                    })
+                ]
+            }
+
+        )
+
+        self.assertEqual(
+            len(
+                self.source.oo_query(
+                    kcidb.orm.Pattern.parse(
+                        '>test[kernelci:kernelci.org:619c656de1fb4af479f2efaa]'
+                        '#<build#'
+                    )
+                ),
+            ),
+            2
+        )
+
+        checkout_of_build_of_test_all_inclusive = self.source.oo_query(
+            kcidb.orm.Pattern.parse(
+                '>test[kernelci:kernelci.org:619c656de1fb4af479f2efaa]#'
+                '<build#<checkout#'
+            )
+        )
+        self.assertEqual(len(checkout_of_build_of_test_all_inclusive), 3)
+        self.assertEqual(
+            checkout_of_build_of_test_all_inclusive['checkout'],
+            [
+                template_dict('checkout', {
+                    "id": "_:kernelci:82bcf49e5e6ad570ff61ffcd"
+                          "210cf85c5ec8d896",
+                    "origin": "kernelci",
+                    "patchset_hash": ""
+                })
+            ]
+        )
+
+        self.assertEqual(
+            self.source.oo_query(
+                kcidb.orm.Pattern.parse(
```

Comment:
You seem to always parse the pattern and then immediately query it, so how about you define a method in KCIDBORMSourceTestCase doing just that. Call it something like `query_str`, meaning "run OO query with a pattern string". Then you get to save two lines of code, and quite a few characters per each query, plus one indent level, making the code considerably shorter and easier to read.